config/initializers/report_designer.rb

```diff
@@ -21,7 +21,17 @@
       user = self.annotator || User.current
       url = self.report_design_field_value('published_article_url')
       language = self.report_design_field_value('language')
-      fields = { user: user, skip_report_update: true , url: url, language: language }
+      state = self.data['state']
+      publisher_id =  state == 'published' ? self.annotator_id : nil
+      fields = {
+        user: user,
+        skip_report_update: true ,
```

Comment:
Minor: It's not related to your changes, but please take this opportunity to delete this space before the comma :)